internal/runners/builds/builds.go

```diff
@@ -192,3 +187,72 @@ func (b *Builds) outputPlain(out *StructuredOutput, fullID bool) error {
 	b.out.Print("\nTo download builds run '[ACTIONABLE]state builds dl <ID>[/RESET]'.")
 	return nil
 }
+
+// getTerminalArtifactMap returns a project's terminal artifact map, depending on the given
+// arguments. By default, the map for the current project is returned, but a map for a given
+// commitID for the current project can be returned, as can the map for a remote project
+// (and optional commitID).
+func getTerminalArtifactMap(
```

Comment:
Let's add a spinner to this function

https://github.com/ActiveState/cli/blob/8bb4da4efcc6dab1b8bff05ffa69ae64162befd9/internal/output/progress.go#L28-L28

Since this often seems to take a few seconds, or even minutes if the build is complex, and otherwise it just hangs.